### ISSUE TITLE
Exclude /usr/lib/python3.*/EXTERNALLY-MANAGED to fix system-wide pip install

### ISSUE THREAD

**Issue by em-** [debuerreotype/debuerreotype#155]
The Python [PEP-668](https://peps.python.org/pep-0668/) introduces a big breaking change by forbidding the system-wide installation of Python packages via `pip`.

All the many many `Dockerfile`s out there running `pip install` are now broken as soon as they try to move to Bookworm-based image.

The goal is to prevent issues when mixing and matching distro-managed packages and pip-installed ones.

However, this is not really an issue for containers. Citing [the use-case #5 in PEP-668 itself](https://peps.python.org/pep-0668/):

> ... builders of base container images may want to ensure that the marker file is not present, even if the underlying OS ships one by default.

The suggestion is thus for Docker containers to remove the `/usr/lib/python3.11/EXTERNALLY-MANAGED` file.

Rather than having everyone `rm` it, could the base container ship a dpkg exclusion to prevent any `/usr/lib/python3.*/EXTERNALLY-MANAGED` files from being installed?

**Comment by yosifkit** [comment]
I don't agree that we should apply this generally in the Debian images. Users of a system installed python should be aware that they are possibly breaking system tools. Users of the `python` images are already using a manually installed, non-distro python so they are unaffected (if they are, then they are using the distro python and ought to be aware that it might be a different version than they expected).

> It is the view of the authors of this PEP that it’s still a good idea to use virtual environments with distro-installed Python interpreters, even in single-application container images. Even though they run a single application, that application may run commands from the OS that are implemented in Python, and if you’ve installed or upgraded the distro-shipped Python packages using Python-specific tools, those commands may break.

**Comment by tianon** [comment]
I'm leaning towards agreeing that this is probably not something we should set for the general case; the workarounds for users to "opt into" the breakage are relatively unobtrusive while still pretty clearly communicating user intent (adding `--break-system-packages` to their `pip install` lines or adding something like `rm -f /usr/lib/python*/EXTERNALLY-MANAGED` to their `RUN` lines somewhere after Python is installed).

My concern is primarily around what the PEP describes as "single-application containers", which is the _ideal_ mode for Docker container images, but I don't think it's fair to say that it's how _everyone_ is using the `debian:xxx` images. :sweat_smile:

**Comment by tianon** [comment]
Oh, this can be set via environment variable too: `ENV PIP_BREAK_SYSTEM_PACKAGES=yes`
